### PR TITLE
Fix Organization global sharing for datasets

### PR DESCRIPTION
**Pull request by anandjakhaniya**
- Fix Organization global sharing for datasets
- Added test cases Org global sharing

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java [hunk 1]
```diff
@@ -1162,7 +1162,6 @@ public void createWorkspaceRoleBinding(
       boolean orgScopedPublic,
       String globalSharing) {
     if (workspaceId != null && !workspaceId.isEmpty()) {
-      Role admin = getRoleByName(roleAdminName, null);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java:1165]: this si just for cleanliness right? or was there a bug you fixed?

File: backend/src/test/java/ai/verta/modeldb/ProjectTest.java [hunk 2]
```diff
@@ -3625,4 +3630,61 @@ public void deleteProjectArtifacts() {
...
+    createProjectRequest =
+        createProjectRequest
+            .toBuilder()
+            .setWorkspaceName(organization.getName())
+            .setProjectVisibility(ProjectVisibility.ORG_SCOPED_PUBLIC)
+            .build();
+    CreateProject.Response createProjectResponse =
+        projectServiceStub.createProject(createProjectRequest);
+    Project project = createProjectResponse.getProject();
+    LOGGER.info("Project created successfully");
```
Comment (ravishetye) [inline backend/src/test/java/ai/verta/modeldb/ProjectTest.java:3693]: can we check if the appropriate role is created as well? same for other test

File: backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java:1165]: it is just moved line where it is actually used. nothing else.

File: backend/src/test/java/ai/verta/modeldb/ProjectTest.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/test/java/ai/verta/modeldb/ProjectTest.java:3693]: Ok. I can add it.

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java [hunk 3]
```diff
@@ -162,7 +162,7 @@ private void createWorkspaceRoleBinding(
           datasetId,
           ModelDBConstants.ROLE_DATASET_ADMIN,
           ModelDBServiceResourceTypes.DATASET,
-          false,
+          datasetVisibility.equals(DatasetVisibility.ORG_SCOPED_PUBLIC),
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java:165]: this is actual bug fix.

File: backend/src/test/java/ai/verta/modeldb/ProjectTest.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/test/java/ai/verta/modeldb/ProjectTest.java:3693]: Added Role check after creating the organization in test cases

**Review by ravishetye — APPROVED** [review]

**Pull request merged**